src/engine/ScreenShotTaker.cs

```diff
@@ -47,24 +65,73 @@ public string TakeAndSaveScreenShot()
         if (error != Error.Ok)
         {
             GD.PrintErr("Saving screenshot failed ", error);
-            return null;
+            return;
         }
 
         GD.Print("Saved screenshot: ", path);
-        return path;
     }
 
     /// <summary>
-    ///   Takes an image of the current viewport
+    ///   Takes and saves a screenshot
     /// </summary>
-    /// <returns>The image</returns>
-    public Image TakeScreenshot()
+    private void TakeScreenshot()
     {
-        var image = GetViewport().GetTexture().GetData();
+        if (isCurrentlyTakingScreenshot)
+        {
+            GD.Print("Already in the process of taking a screenshot.");
+            return;
+        }
 
-        // TODO: do we always need this?
-        image.FlipY();
+        isCurrentlyTakingScreenshot = true;
+        FileHelpers.MakeSureDirectoryExists(Constants.SCREENSHOT_FOLDER);
 
-        return image;
+        // If ScreenFilter is active, turn it off before taking a screenshot.
+        if (ColourblindScreenFilter.Instance.Visible)
+        {
+            step = Steps.Start;
+            ScreenFilterScreenshotStepper();
+            return;
+        }
+
+        SaveScreenshot(GetViewportTextureImage());
+        isCurrentlyTakingScreenshot = false;
+    }
+
+    /// <summary>
+    ///   Invokes itself to:
+    ///   1: Hide the ScreenFilter and wait a frame.
+    ///   2: Wait another frame.
+    ///   3: Take the screenshot, show the filter and wait a frame so the filter can show itself again faster.
+    ///   4: Save the screenshot.
+    /// </summary>
+    private void ScreenFilterScreenshotStepper()
+    {
+        switch (step)
+        {
+            case Steps.Start:
+                GD.Print("Start");
+                ColourblindScreenFilter.Instance.Hide();
+                step = Steps.Wait;
+                break;
+            case Steps.Wait:
+                GD.Print("Wait");
+                step = Steps.TakeScreenshot;
+                break;
+            case Steps.TakeScreenshot:
+                GD.Print("TakeSceenshot");
+                screenshotImage = GetViewportTextureImage();
+                ColourblindScreenFilter.Instance.Show();
+                step = Steps.Save;
+                break;
+            case Steps.Save:
```

Comment:
Why is the image not written to disk when it was actually taken? Instead an extra frame is waited?

Reply:
if I save it directly the blink is much longer, as the save blocks the thread until save finished.
Saving in the next frame has the ScreenFilter on again.

Reply:
See my previous comment about the extra frame, so I guess it would be fine. What about writing the file to disk from a background task in TaskExecutor? I would expect it to be a thread safe operation of writing the file to disk.

Reply:
great idea, i try that.

Reply:
changed the last step to a background task.